### PR TITLE
fix: 🐛 enforce lower and upper file size limit when triggering analysis [ROAD-799][ROAD-814]

### PR DESCRIPTION
**Pull request by bastiandoetsch**
also increase retries for upload to 10

### CODE REVIEW

File: src/main/java/ai/deepcode/javaclient/core/AnalysisDataBase.java [hunk 1]
```diff
@@ -234,6 +235,7 @@ public void updateCachedResultsForFiles(
           allProjectFiles.stream()
               .filter(Objects::nonNull)
               .filter(file -> !mapFile2Suggestions.containsKey(file))
+              .filter(file -> pdUtils.getFileSize(file) > 0 && pdUtils.getFileSize(file) < MAX_FILE_SIZE)
```
Comment (ArtsiomCh) [inline src/main/java/ai/deepcode/javaclient/core/AnalysisDataBase.java:238]: That check should be not added here but fixed at  `ai.deepcode.javaclient.core.DeepCodeUtilsBase#isSupportedFileFormat` as we need such check in other places too (not while analyzing)

**Review by ArtsiomCh — CHANGES_REQUESTED** [review]

File: src/main/java/ai/deepcode/javaclient/core/AnalysisDataBase.java (hunk 1 above)
Comment (bastiandoetsch) [reply src/main/java/ai/deepcode/javaclient/core/AnalysisDataBase.java:238]: Hm, I thought it was the entry point to any analysis, so filtering there seems to make sense. Do we have other entry points I'm not aware of. I considered using `isSupportedFileFormat` and would even consider putting it there, too - if my understanding is correct about entry points into the code analysis.

Comment (bastiandoetsch) [reply src/main/java/ai/deepcode/javaclient/core/AnalysisDataBase.java:238]: But yes, you are correct - it is better to have it as you say - just looked again!

Comment (bastiandoetsch) [reply src/main/java/ai/deepcode/javaclient/core/AnalysisDataBase.java:238]: Incoming commit ;)

**Comment by ArtsiomCh** [comment]
We forgot changelog update

**Comment by bastiandoetsch** [comment]
Can't merge due to Code environment :(

**Pull request merged**